lapce-data/src/buffer.rs

```diff
@@ -1184,6 +1184,20 @@ impl Buffer {
             line
         }
     }
+
+    // The folder function calculates offset of the next/previous word boundary
+    // for the given offset.
+    fn move_by_word<F>(
+        &self, offset: usize, count: usize, tab_width: usize, folder: F
+    ) -> (usize, ColPosition)
+    where
+        F: FnMut(usize, ()) -> usize {
+        let new_offset = std::iter::repeat(()).take(count).fold(offset, folder);
```

Comment:
This is a bit cryptic to eyes not accustomed to functional primitives like fold. I think a comment might be helpful.

Reply:
Will try.